python/UniverseWrapper.cpp

```diff
@@ -311,23 +311,29 @@ namespace FreeOrionPython {
                 +[](const std::string& name) -> bool { return GetValueRef<int>(name, true); },
                 "Returns true/false (boolean) whether there is a defined int-valued "
                 "scripted constant with name (string).");
-        py::def("getNamedValue",
-                +[](const std::string& name) -> py::object {
-                    auto eval = [](auto&& ref) -> py::object {
+        py::def("getNamedReal",
+                +[](const std::string& name) -> double {
+                    if (const auto ref = GetValueRef<double>(name, true)) {
                         if (ref->ConstantExpr()) 
-                            return py::object(ref->Eval());
+                            return ref->Eval();
                         const ScriptingContext context;
-                        return py::object(ref->Eval(context));
-                    };
-
-                    if (const auto ref = GetValueRef<double>(name, true))
-                        return eval(ref);
-                    else if (const auto ref = GetValueRef<int>(name, true))
-                        return eval(ref);
-                    else
-                        return py::object();
+                        return ref->Eval(context);
+                    }
+                    return 0.0;
```

Comment:
```suggestion
                    return py::object();
```

Reply:
Actually that does not work:
`error: cannot convert ‘boost::python::api::object’ to ‘double’ in return`

It almost works with:
```
        py::def("getNamedReal",
                +[](const std::string& name) -> boost::optional<double> {
                    if (const auto ref = GetValueRef<double>(name, true)) {
                        if (ref->ConstantExpr()) 
                            return ref->Eval();
                        const ScriptingContext context;
                        return ref->Eval(context);
                    }
                    return boost::optional<double>(boost::none);
                },
```

It generates the interface, but when I run it, I get an exception:
`No to_python (by-value) converter found for C++ type: boost::optional<int>`

Reply:
I guess the lambda would need to have the signature `[](const std::string& name) -> py::object` as before then? Don't think you need `boost::optional` to get the equivalent functionality on the Python side of the API.

Reply:
Well, using return type py::object would surely work, but defeats the purpose. Okay, I'll get rid of the warning, but PyCharm would still only see two functions that returns an object and complain when I use it like a number.

Well, I guess I could use a wrapper in python, that casts the value to the correct type...

Reply:
an alternative is to have separate `IsNamedReal` function to query before using, and to return 0 if there isn't one, as you've tried to do.

Reply:
Well, doing this would also require wrapper functions.
I've now done it with return value py::object and wrapper functions in freeorion_tools.

Reply:
> an alternative is to have separate `IsNamedReal` function to query before using, and to return 0 if there isn't one, as you've tried to do, and then update the documentation to not claim it returns `none` if there isn't.

Since I started using wrappers anyway, I've gone that way.